src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String OPERATION_TYPE_BUY = "buy";
+    private static final String OPERATION_TYPE_SUPPLY = "supply";
+    private static final String COMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder report = new StringBuilder();
+        List<String> stringFromFile = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String read = bufferedReader.readLine();
+            while (read != null) {
+                stringFromFile.add(read);
+                read = bufferedReader.readLine();
+            }
+            buildReport(stringFromFile, report);
+
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to read from file: " + fromFileName, e);
+        }
+        writeToFile(toFileName, report);
+    }
+
+    private static void buildReport(List<String> stringFromFile, StringBuilder report) {
+        int buy = 0;
+        int supply = 0;
+
+        for (String record : stringFromFile) {
+            String[] recordParts = record.split(COMA);
+            int digit = Integer.parseInt(recordParts[1]);
```

Comment:
Instead of using `split()` directly, it is recommended to handle potential exceptions that may occur if the input is malformed. For example, if the record does not have a comma or has more than two parts, this could lead to `ArrayIndexOutOfBoundsException` or incorrect data parsing.